src/containers/sb-file-uploader.jsx

```diff
@@ -79,10 +79,13 @@ class SBFileUploader extends React.Component {
             reader.readAsArrayBuffer(thisFileInput.files[0]);
             // extract the title from the file and set it as current project title
             if (thisFileInput.files[0].name) {
-                const matches = thisFileInput.files[0].name.match(/^(.*)\.sb3$/);
+                const matches = thisFileInput.files[0].name.match(/^(.*?)(\.sb[23]?)?$/);
                 if (matches) {
                     const truncatedProjectTitle = matches[1].substring(0, 100);
-                    this.props.onSetProjectTitle(truncatedProjectTitle);
+                    this.props.onSetReduxProjectTitle(truncatedProjectTitle);
```

Comment:
When do we not have `props.onUpdateProjectTitle`? This was set up this way before but now It's apparent that this is a little hacky — we're setting it in redux and then asking something different to also update the title to stay in sync. The best way to stay in sync is to always update at the top and let the state go down from there.

Reply:
It's a little hacky. Maybe gui.jsx could listen for changes to the redux title and call onUpdateProjectTitle, but it's already listening for changes to the incoming projectTitle prop and using that to set the redux project title... would it work to do both in the same place?

Reply:
We shouldn't need to listen for changes to the redux title, `onUpdateProjectTitle` should result in a change to the incoming `title` prop. That would then should cause GUI to update the redux state. Then everything that uses the redux title should read it from there. If `onUpdateProjectTitle` doesn't result in a change to the incoming `title` prop then we don't want the redux state to be different than that prop.

We can't assume what will happen when we call `onUpdateProjectTitle` — maybe there's a sanitizer before it updates the title, or maybe the title shouldn't change in a certain context.